src/app/dashboard/dashboard.component.ts

```diff
@@ -1,8 +1,35 @@
-import { Component, OnInit } from '@angular/core';
+import { Component } from '@angular/core';
+import { ApiService } from '../Services/api.service';
+import { HttpErrorResponse } from '@angular/common/http';
 
 @Component({
   selector: 'app-dashboard',
   templateUrl: './dashboard.component.html',
   styleUrls: ['./dashboard.component.css'],
 })
-export class DashboardComponent {}
+export class DashboardComponent {
+  constructor(private apiService: ApiService) {}
+  errorClockIn: number = null;
+  errorClockOut: number = null;
```

Comment:
1. Pola w komponencie definiujemy powyżej constructora,
2. Null to domyślna wartość, jeśli niczego nie przypisujemy do danego pola, dlatego wystarczy zapisać to w taki sposób: 

```suggestion
  errorClockIn: number;
  errorClockOut: number;
  
  constructor(private apiService: ApiService) {}
```